### ISSUE TITLE
Hostnames can't start with a range in inventory as of 2.0.1.0

### ISSUE THREAD

**Issue by Excedrin** [ansible/ansible#15331]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible 2.0.1.0
```
##### CONFIGURATION

```
% cat ansible.cfg
[defaults]
hostfile = inventory

% cat inventory
[test]
[1:2].fail
fail.[1:2]
fail.[1:2].fail
```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

The fix for this issue breaks hosts that start with a range:
https://github.com/ansible/ansible/issues/13917
This commit:
https://github.com/ansible/ansible/commit/a7dd425620e0a40911908a41e1b378183ae690f4
##### STEPS TO REPRODUCE

setup config and inventory as shown above, then run:

```
ansible test --list-hosts

```

##### EXPECTED RESULTS

```
ansible test --list-hosts
  hosts (6):
    1.fail
    2.fail
    fail.1
    fail.2
    fail.1.fail
    fail.2.fail
```
##### ACTUAL RESULTS

```
ansible test --list-hosts
ERROR! inventory:2: Invalid section entry: '[1:2].fail'. Please make sure that there are no spacesin the section entry, and that there are no other invalid characters

```

Changing line 127 in ansible/inventory/ini.py to the following might fix both issues, since ranges have to have something either before or after or both for them to be useful. In other words, if it starts and ends with square brackets and doesn't match the section regex, then it is an actual error.

```
            elif line.startswith('[') and line.endswith(']'):
```

**Comment by e-alfred** [comment]
Hello,

I tried the change in the ini.py file but the error persists:

`ansible-playbook -vvvvvv playbook.yml 
Using /etc/ansible/ansible.cfg as config file
ERROR! /etc/ansible/hosts:39: Invalid section entry: '[Serverlist]'. Please make sure that there are no spacesin the section entry, and that there are no other invalid characters`

I am running 2.0.1 as well.

**Comment by Excedrin** [comment]
It looks like your use case is different from mine. The problem that this bug and fix addresses are cases where an Ansible range starts the line. Like "^[1:2].fail$" (with ^ and $ added as start and end of line).

Your error doesn't include a range at all, and I suspect that you have some unicode characters or something invalid before line 39. If you add part of your /etc/ansible/hosts I'll take a look at it.

**Comment by e-alfred** [comment]
Well, if I comment out the lines with brackets everything works perfectly. I upgraded from 2.0.0 where everything worked fine with the same hosts file.

Here is a section copied from a freshly created hosts file, the error still persists:

```
server1.subdomain.domain.org

[Serverlist]
myserver.subdomain.domain.org
```

**Comment by Excedrin** [comment]
I setup ansible.cfg the same as earlier, with inventory/hosts containing exactly what you showed above and got the following results:

% ansible --version
ansible 2.0.1.0
  config file = /home/test/ansible_db/ansible.cfg
  configured module search path = Default w/o overrides

% ansible all --list-hosts
  hosts (2):
    server1.subdomain.domain.org
    myserver.subdomain.domain.org

So I'm still not sure what's causing the error in your case, I'd need more details to be able to reproduce it.

**Comment by Excedrin** [comment]
abadger's fix works, can this please get merged, it's really annoying, every time a new version comes out my inventory breaks.

**Comment by abadger** [comment]
Merged to devel and the stable-2.1 tree.  Not sure yet if it will be in the 2.1.2 release.  That depends on whether we have other fixes that require a 2.1.2rc2.
